ethers-middleware/src/signer.rs

```diff
@@ -131,12 +131,12 @@ where
         // compare chain_id and use signer's chain_id if the tranasaction's chain_id is None,
         // return an error if they are not consistent
         let chain_id = self.signer.chain_id();
+        let mut tx = tx.clone();
```

Comment:
do we need to clone? let's just make `mut tx` instead in the function declaration?

Reply:
oh right, yeah we can do that, that makes more sense